test/e2e/middleware-matcher/index.test.ts

```diff
@@ -217,6 +217,49 @@ describe('using a single matcher', () => {
   })
 })
 
+describe('using root matcher', () => {
```

Comment:
Can we add a test case to ensure https://github.com/vercel/next.js/issues/39396 is working as expected?

Reply:
I added the test cases, but the first iteration of this PR did not fixed the bug report.
This is because we're having an extra / in the regexp when passing `trailingSlash: true`:

![image](https://user-images.githubusercontent.com/186268/183472849-45190a01-7b26-4ee0-83cf-0134ded9fb79.png)

Hence why I removed the special handling of `trailingSlash` within `getMiddlewareRegExpStrings()`